### ISSUE TITLE
Events endpoint?

### ISSUE THREAD

**Issue by friggeri** [appdotnet/api-spec#92]
I've been thinking of doing some graph analysis of the growth of the app.net, but there are key pieces of information which are missing. For example, I think it would be really useful to have an `events` endpoint which would list the interactions of the user with the service (you can think of this as an open-log).

For example:

```
GET /stream/0/events/[user_id]/followed
```

Would return an array of all the times the user has followed someone else:

```
{
  "id": "1",
  "user":  { ... },
  "event_type":"followed",
  "followed_id":"2",
  "followed_user":  { ... },
  "time": "2012-07-16T17:25:47Z"
}
```

Similarly, `GET /stream/0/events/[user_id]/was_followed` would return the list of all events pertaining to other users following this user, `GET /stream/0/events/[user_id]/auth` would list all instances where the user was authentified along with information about the app he used, `GET /stream/0/events/[user_id]` would return a superset of the previous which contains all logged events for this user, `GET /stream/0/events` would return all events for all users, etc.

Use cases: 
- First and foremost, this would allow to list changes/evolution of followers (useful in terms of analytics)
- Second, this would bring more granularity as well as a queryable unified endpoint to be used in notifications 
- Third, as a social network analyst I'd love to be able to leverage this information to try and model the growth of the followers graph

Basically, my idea is that the value I add as a user is not limited to the data and posts I put into the service, but extends to my interactions with the service. As such, these interactions should be accessible and queryable.

Would love to hear your thoughts.

**Comment by abraham** [comment]
This could be interesting.

**Comment by kosso** [comment]
+1 to this.  I do a similar thing for 'activity streams'.  It also makes things like useful push notifications very easy.

**Comment by mthurman** [comment]
This seems like a perfect use case for [subscriptions](https://github.com/appdotnet/api-spec/blob/master/resources/subscriptions.md). I get that subscriptions are real-time and don't provide historical data but I think they would provide most of what's needed here for things like push notifications.

You could do analysis with that subscriptions, it just wouldn't be historical (which may be the whole point of the original issue). I guess my point is, we've already got things in the api that enable like push notifications.

**Comment by friggeri** [comment]
You got it right, the whole point of the issue is to provide accesse to historical data. This would help for analysis on the long term and as I said earlier, the interactions of a user with the service is also a part of his "data", in an extended sense.
